### PR TITLE
Command mode help text

### LINKED ISSUE #76: Command Mode specific help text
# Goal

Right now, the command `:help` prints the regular help text for normal mode controls i.e. the same as the `?` key. Instead, it should print help specific to command mode. 

# Implementation

* Using the text already in the readme, make a new help text method similar to the existing one, but for command mode. 
* On command `:help` print this help text instead of the existing help text. 
* Instead of a boolean for render_help in `ui::State`, make an enum, with three variants...
  * None
  * Normal
  * Command
or something like that.

### PR DESCRIPTION
**Pull request by Davejkane**
This PR closes [#76](https://github.com/Davejkane/riv/issues/76)

### CODE REVIEW

**Comment by Davejkane** [comment]
@gurgalex sorry for all the tiny code review requests.

File: src/program/command_mode.rs [hunk 1]
```diff
@@ -327,9 +327,12 @@ impl<'a> Program<'a> {
                 }
                 self.newglob(&arguments);
             }
-            Commands::Help => {
-                self.ui_state.render_help = !self.ui_state.render_help;
-            }
+            Commands::Help => match self.ui_state.render_help {
+                HelpRender::None | HelpRender::Normal => {
```
Comment (gurgalex) [inline src/program/command_mode.rs]: Would this be expressed clearly with `_`

If the HelpRender is `Command`. Set it to None
Else Render the `Command` helpbox
_

File: src/program/render.rs [hunk 2]
```diff
@@ -384,3 +384,26 @@ fn help_text() -> Vec<&'static str> {
...
     ]
 }
+
+fn command_help_text() -> Vec<&'static str> {
+    vec![
+        "+-------+------------+----------+-------------------------------------+",
+        "| Short | Long       | Argument | Action                              |",
+        "+-------+------------+----------+-------------------------------------+",
+        "| ng    | newglob    | Required | The new glob/directory/file         |",
+        "| ?     | help       | None     | Toggle help box                     |",
```
Comment (gurgalex) [inline src/program/render.rs:394]: Should pressing `?` with the command mode help box open, but back in Normal mode, open up the Normal mode help box, or just close the Command mode box?

It feels nice, but unexpected at first.

Steps:
1. Open Command mode
2. Enter `?` as a command

Command help box is now open.

3. The use is back in Normal mode.
4. Press ?.
The Normal mode help box is displayed.

5. Press ? again
The normal mode help box is gone.

No more help boxes are visible.

Comment (gurgalex) [inline src/program/render.rs:394]: On an invalid command. [Let the user know that help is available.](https://github.com/Davejkane/riv/blob/029464844ccab490840851af2cb784ff5a524848/src/program/command_mode.rs#L65)

Maybe something like
`No such command "{}". Type "?" for help`

File: src/ui.rs [hunk 3]
```diff
@@ -143,7 +154,12 @@ pub fn process_normal_mode<'a>(state: &mut State<'a>, event: &Event) -> Action<'
             "g" => state.process_action(Action::First),
             "G" => state.process_action(Action::Last),
             "?" => {
-                state.render_help = !state.render_help;
+                match state.render_help {
+                    HelpRender::None | HelpRender::Command => {
```
Comment (gurgalex) [inline src/ui.rs]: `_` match for HelpRender's that are not `Normal`

**Review by gurgalex — CHANGES_REQUESTED** [review]
@Davejkane A welcome improvement to knowing all the commands in Command mode.

I do think it'd be useful to update the Invalid command dialog to reference that a help box is available with ?.

File: src/program/render.rs (hunk 2 above)
Comment (Davejkane) [reply src/program/render.rs:394]: Yes, for me this was the intended behaviour and I'm happy with it.

Comment (gurgalex) [reply src/program/render.rs:394]: I'm happy with it as well. Just making sure.

**Comment by Davejkane** [comment]
@gurgalex, concerns addresses. Please re-review

**Review by gurgalex — APPROVED** [review]

**Comment by gurgalex** [comment]
All good. Ready for merge.

**Pull request merged**